### DISCUSSION TITLE
Atoms with a selector default cause memory leak

### DISCUSSION THREAD

**Discussion by janicduplessis** [facebookexperimental/Recoil#1821] [General]
### Issue

Creating an atom with a selector to lazily initialize the default value like mentioned in the documentation [here](https://recoiljs.org/docs/guides/asynchronous-data-queries#query-default-atom-values) will cause all values set to it to leak.

This will leak:

```js
atom({
  key: "transactions",
  default: selector({
    key: "transactions/default",
    get: retrieveTransactions
  })
});
```

This is fine:

```js
atom({
  key: "transactions",
  default: retrieveTransactions()
});
```

The issue is caused by the selector created by atomWithFallback (https://github.com/facebookexperimental/Recoil/blob/main/packages/recoil/recoil_values/Recoil_atom.js#L661) which will cause all set on the atom to be cached. 

### Suggestion

Disabling cache for the selector created by `atomWithFallback` fixes the problem and should not cause any issue as I don't see why we would want to keep old values for this selector. It is also very cheap to execute.

```diff
   const sel = Recoil_selector({
     key: `${options.key}__withFallback`,
+    cachePolicy_UNSTABLE: { eviction: 'most-recent' },
     get: ({
       get
     }) => {
```

If this makes sense I can open a PR to make this change.

### Repro

https://codesandbox.io/s/clever-wilbur-8cv6qr?file=/src/App.js

Open chrome devtools and go to the Memory tab. Notice ever increasing memory usage for the .csb.app js context.

![Screen Shot 2022-05-24 at 8 28 36 PM](https://user-images.githubusercontent.com/2677334/170153396-15890f6e-9a73-4f8f-9a5e-542bf7167b1d.png)

By taking a snapshot we can clearly see the leaked Transaction objects.

![Screen Shot 2022-05-24 at 8 30 10 PM](https://user-images.githubusercontent.com/2677334/170153481-bc59ebdb-589a-4571-bbe3-09eb38a95f65.png)

Set `LEAK` to false and refresh

Observe no more memory leak

**Comment by noritheduck** [comment]
Hey @janicduplessis, have you checked out Least Recently Used (LRU) cache in the selector documentation? https://recoiljs.org/docs/api-reference/core/selector/

**Reply by janicduplessis** [reply]
@noritheduck I think this is actually an issue since it is impossible to configure the selector created internally by atomWithFallback. See https://github.com/facebookexperimental/Recoil/blob/main/packages/recoil/recoil_values/Recoil_atom.js#L661. The selector passed as default is fine and is not the one causing the problem.

**Reply by noritheduck** [reply]
We re-opened the issue here https://github.com/facebookexperimental/Recoil/issues/1840. We are basically disabling cache in the wrapping selector for the atomWithFallback. Thank you @janicduplessis.
